pkg/cmd/roachtest/tests/db_console_endpoints.go

```diff
@@ -0,0 +1,419 @@
+	c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings())
+
+	if err := initializeSchemaAndIDs(ctx, c, t.L()); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := testEndpoints(ctx, c, t.L(), getEndpoints(t), false); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func testEndpoints(
+	ctx context.Context,
+	c cluster.Cluster,
+	l *logger.Logger,
+	endpoints []endpoint,
+	multiVersionTest bool,
+) error {
+	// Get the NodeIDs and DB console URLs for each node.
+	idMap := make(map[int]roachpb.NodeID)
+	urlMap := make(map[int]string)
+	adminUIAddrs, err := c.ExternalAdminUIAddr(ctx, l, c.CRDBNodes())
+	if err != nil {
+		return err
+	}
+	client := roachtestutil.DefaultHTTPClient(c, l, roachtestutil.HTTPTimeout(15*time.Second), roachtestutil.WithHeaders(map[string]string{"X-Cockroach-API-Session": "cookie"}))
+	for i, addr := range adminUIAddrs {
+		var details serverpb.DetailsResponse
+		url := `https://` + addr + `/_status/details/local`
+		if err := retry.ForDuration(10*time.Second, func() error {
+			return client.GetJSON(ctx, url, &details)
+		}); err != nil {
+			return err
+		}
+		idMap[i+1] = details.NodeID
+		urlMap[i+1] = `https://` + addr
+	}
+
+	// Test each endpoint.
+	for _, ep := range endpoints {
+		// For endpoints with node ID placeholders, test with "local", own node ID, and another node ID.
+		if ep.hasNodeID() {
+			for nodeID := 1; nodeID <= len(idMap); nodeID++ {
+				baseURL := urlMap[nodeID]
+				// Test with "local".
+				if err := testEndpoint(ctx, client, baseURL, ep, "local", ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s with local", ep.url)
+				}
+				// Test with own node ID.
+				if err := testEndpoint(ctx, client, baseURL, ep, idMap[nodeID].String(), ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s with own node ID", ep.url)
+				}
+				// Test with another node ID.
+				otherNodeID := (nodeID % len(idMap)) + 1
+				if err := testEndpoint(ctx, client, baseURL, ep, idMap[otherNodeID].String(), ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s with other node ID", ep.url)
+				}
+			}
+		} else {
+			// For endpoints without node IDs, test on each node.
+			for nodeID := 1; nodeID <= len(idMap); nodeID++ {
+				if err := testEndpoint(ctx, client, urlMap[nodeID], ep, "", ep.getVerifyResponse(), l, multiVersionTest); err != nil {
+					return errors.Wrapf(err, "failed testing endpoint %s", ep.url)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func testEndpoint(
+	ctx context.Context,
+	client *roachtestutil.RoachtestHTTPClient,
+	baseURL string,
+	ep endpoint,
+	nodeID string,
+	verifyResponse func(*http.Response) error,
+	l *logger.Logger,
+	multiVersionTest bool,
+) error {
+	fullURL := baseURL + ep.url
+	var err error
+	fullURL, err = fillPlaceholders(fullURL, nodeID)
```

Comment:
I'm leaning towards this file being broken into two separate files:
* One for collecting, setting up the tests (parsing JSON, initializing the cluster, injecting the parameter values)
* One for test execution (constructing the urlmap, calling the endpoints, verifying the response)

I feel like these could all live in the new `roachtest/tests/db-console` directory you created. What are your thoughts?